CMakeLists.txt

```diff
@@ -250,15 +253,57 @@ if(${UseProb3ppLinear} EQUAL 1)
 endif()
 
 if(${UseNuFASTLinear} EQUAL 1)
-    CPMAddPackage(
-      	NAME NuFAST
-      	GIT_TAG main
-      	GITHUB_REPOSITORY PeterDenton/NuFast
-      	DOWNLOAD_ONLY
-      )
-      target_compile_definitions(NuOscillatorCompilerOptions INTERFACE UseNuFASTLinear=1)
+  CPMAddPackage(
+    NAME NuFAST
+    GIT_TAG main
+    GITHUB_REPOSITORY PeterDenton/NuFast
+    DOWNLOAD_ONLY
+  )
+  target_compile_definitions(NuOscillatorCompilerOptions INTERFACE UseNuFASTLinear=1)
 endif()
 
+if(${UseNuSQUIDSLinear} EQUAL 1)
+  find_package(GSL REQUIRED)
+  find_package(HDF5 REQUIRED)
+
+  include(ExternalProject)
+
+  ExternalProject_Add(SQUIDS
+    GIT_REPOSITORY https://github.com/jsalvado/SQuIDS.git
+    GIT_TAG master
+    BINARY_DIR = <SOURCE_DIR> 
+    CONFIGURE_HANDLED_BY_BUILD true
+    CONFIGURE_COMMAND <SOURCE_DIR>/configure --with-gsl=${GSL_ROOT_DIR} --prefix=<INSTALL_DIR>
+    BUILD_COMMAND make
+    INSTALL_COMMAND make install
+  )
+
+  SET(SQUIDS_INC_DIR ${CMAKE_CURRENT_BINARY_DIR}/SQUIDS-prefix/include)
+  SET(SQUIDS_LIB_DIR ${CMAKE_CURRENT_BINARY_DIR}/SQUIDS-prefix/lib)
+
+  add_library(SQUIDS::SQUIDS SHARED IMPORTED)
+  set_target_properties(SQUIDS::SQUIDS PROPERTIES IMPORTED_LOCATION ${SQUIDS_LIB_DIR}/libSQuIDS.so)
+  
+  ExternalProject_Add(NUSQUIDS
+    GIT_REPOSITORY https://github.com/arguelles/nuSQuIDS.git
+    GIT_TAG master
+    BINARY_DIR = <SOURCE_DIR>
```

Comment:
nuSQuIDS is actually being developed. Is it worth to use tag etc. instead of master?

Reply:
The tagging system being used by NuSQUIDS is not clear.. the tags are much older than recent developments, some of which are required for the planned quantum decoherence studies planned. It might be worth pulling an individual git commit hash if that's possible.